src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/IResolverUICommunicator.cs

```diff
@@ -23,14 +23,15 @@ public interface IResolverUICommunicator
     /// Sends request for the shell to show a window for the user to select the wished intent for resolving the RaiseIntentForContext.
     /// </summary>
     /// <param name="intents"></param>
+    /// <param name="timeout"></param>
     /// <returns></returns>
-    public Task<ResolverUIIntentResponse?> SendResolverUIIntentRequest(IEnumerable<string> intents, CancellationToken cancellationToken = default);
+    public Task<ResolverUIIntentResponse?> SendResolverUIIntentRequest(IEnumerable<string> intents, TimeSpan? timeout = null);
```

Comment:
QQ: Why it's better to have a TimeSpan instead of the CancellationToken?

Reply:
It seemed overkill to have to create the cancellationtokensource in the DesktopAgent just for a timeout, so I moved creating the token to the implementation. Maybe there was more thought to it but I don't remember it and certainly didn't act on it. I can revert this change as it's not really adding anything to the changes.